spki/src/traits.rs

```diff
@@ -130,3 +130,39 @@ where
         }
     }
 }
+
+/// Returns `AlgorithmIdentifier` associated with the signature system.
+///
+/// Unlike AssociatedAlgorithmIdentifier this is intended to be implemented for public and/or
+/// private keys.
+pub trait SignatureAlgorithmIdentifier {
+    /// Algorithm parameters.
+    type Params: der::Encode;
+
+    /// `AlgorithmIdentifier` for the corresponding singature system.
+    const ALGORITHM_IDENTIFIER: AlgorithmIdentifier<Self::Params>;
```

Comment:
This is ambiguous with `AssociatedAlgorithmIdentifier`. Perhaps this?
```suggestion
    const SIGNATURE_ALGORITHM_IDENTIFIER: AlgorithmIdentifier<Self::Params>;
```

Reply:
Ack, I committed both suggestions.